### PR TITLE
Oneview_logical_downlink API 300 Support for Synergy and C7000

### LINKED ISSUE #14: Logical Downlink API300 Support
### Scenario/Intent
Use Oneview_logical_downlink with API 300 on the Synergy and C7000 hardware variants

### Environment Details
 - **Puppet module for HPE OneView Version:** 2.0.0
 - **OneView SDK Version:** 3.0.0+
 - **OneView Appliance Version:** 3.0.0
 - **OneView Client API Version:** 300
 - **Puppet Version:** 4.0.0+
 - **Ruby Version:** 2.2.6
 - **Platform:** Windows/CentOS

### Steps to Reproduce
1- Create a login file with api_version = 300 and hardware_variant = "C7000" or "Synergy"
2- Create a manifest in which I call the Oneview_logical_downlink resource

### Expected Result
Should use the API 300 of the chosen hardware variant to perform any actions

### Actual Result
Uses the API 200 of the C7000 hardware variant to perform all actions.

### PR DESCRIPTION
**Pull request by fgbulsoni**
### Description
Expands support to API 300 for the Oneview_logical_downlink resource

### Issues Resolved
#14 

### Check List
- [X] New functionality includes testing.
  - [X] All tests pass (`$ rake test`).
- [X] New functionality has been documented in the README if applicable.
  - [X] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [X] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: spec/unit/provider/oneview_logical_downlink_c7000_spec.rb [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+      allow(resourcetype).to receive(:find_by).and_return([test])
+      provider.exists?
+    end
+
+    it 'should raise error when running get without ethernet' do
+      expect { provider.get_without_ethernet }.to raise_error(/The method #get_without_ethernet is unavailable for this resource/)
+    end
+
+    it 'should raise error when running get without ethernet - Test disabled due to bug on oneview-sdk'
+    # resource['data'] = {}
```
Comment (abiliogp) [inline spec/unit/provider/oneview_logical_downlink_c7000_spec.rb]: Why these code are commented?

File: spec/unit/provider/oneview_logical_downlink_synergy_spec.rb [hunk 2]
```diff
@@ -50,31 +59,38 @@
...
-      allow(resourcetype).to receive(:get_without_ethernet).with(anything).and_return(test)
-      expect(provider.get_without_ethernet).to be
     end
+
+    it 'should raise error when running get without ethernet' do
+      expect { provider.get_without_ethernet }.to raise_error(/The method #get_without_ethernet is unavailable for this resource/)
+    end
+
+    it 'should raise error when running get without ethernet - Test disabled due to bug on oneview-sdk'
+    # resource['data'] = {}
```
Comment (abiliogp) [inline spec/unit/provider/oneview_logical_downlink_synergy_spec.rb]: Why these code are commented?

**Review by abiliogp — CHANGES_REQUESTED** [review]
I found some codes commented out.
Please, fix that code.

File: spec/unit/provider/oneview_logical_downlink_c7000_spec.rb (hunk 1 above)
Comment (fgbulsoni) [reply spec/unit/provider/oneview_logical_downlink_c7000_spec.rb]: 'Test disabled due to bug on oneview-sdk'

File: spec/unit/provider/oneview_logical_downlink_synergy_spec.rb (hunk 2 above)
Comment (fgbulsoni) [reply spec/unit/provider/oneview_logical_downlink_synergy_spec.rb]: 'Test disabled due to bug on oneview-sdk' - Should be re-enabled/uncommented once oneview-sdk is fixed

**Comment by tmiotto** [comment]
Maybe we could use `xit` instead commenting out so we don't lose track of it

**Comment by fgbulsoni** [comment]
Commented out test changed for xit. @abiliogp please verify.

**Review by abiliogp — APPROVED** [review]

**Pull request merged**